### PR TITLE
default to production

### PR DESCRIPTION
**Pull request by jmervine**
Addressing comments on https://github.com/MaxCDN/bootstrap-cdn/pull/706 around production vs development env.

@XhmikosR

### CODE REVIEW

**Comment by XhmikosR** [comment]
I don't think this is needed, at all.

I mean, can't we just set the env var in make.js for start target to production? I tried your branch with `node make start` and I still get the unminified Jade output.

**Comment by jmervine** [comment]
If you're still getting the minified html with this branch, then you must have `NODE_ENV=development` set. 

That said, if I pass `production` to start js, you'll get https redirection and won't be able to hit the page.

**Comment by XhmikosR** [comment]
No, I'm getting the unminifed HTML even with `node make start`.

**Comment by jmervine** [comment]
Okay, yeah... oops... I forgot I was forcing unset `NODE_ENV` to `development` up top. I reverted the bulk of this and it should be working now.

I still need to force the https redirection off unless configured otherwise to ensure that it works locally, but otherwise...

**Comment by XhmikosR** [comment]
Now `node make start` is using minified HTML but so does `npm run run`.

I still believe it's **a lot** simpler to just set the env var in make.js for the `start` target...

**Comment by jmervine** [comment]
@XhmikosR how's this?

File: app.json [hunk 1]
```diff
@@ -1,6 +1,11 @@
 {
-  "name":"bootstrap-cdn",
-  "scripts":{},
-  "env":{},
-  "addons":[]
+    "name":"bootstrap-cdn",
```
Comment (XhmikosR) [inline app.json]: While at it, maybe add a space before `:`.

**Comment by XhmikosR** [comment]
Much better although I'm getting an error when running `make stop`

```
C:\Users\xmr\Desktop\bootstrap-cdn>node make start
warn:    --minUptime not set. Defaulting to: 1000ms
warn:    --spinSleepTime not set. Your script will exit if it does not stay up for at least 1000ms
info:    Forever processing file: app.js
warn:    --minUptime not set. Defaulting to: 1000ms
warn:    --spinSleepTime not set. Your script will exit if it does not stay up for at least 1000ms
info:    Forever processing file: app.js

C:\Users\xmr\Desktop\bootstrap-cdn>node make stop
error:   Forever cannot find process with id: app.js
C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\shelljs\src\common.js:340
        throw e;
        ^

Error: exec: internal error
    at Object.error (C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\shelljs\src\common.js:67:11)
    at _exec (C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\shelljs\src\exec.js:266:12)
    at C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\shelljs\src\common.js:275:23
    at assertExec (C:\Users\xmr\Desktop\bootstrap-cdn\make.js:26:16)
    at Function.target.stop (C:\Users\xmr\Desktop\bootstrap-cdn\make.js:70:9)
    at Object.global.target.(anonymous function) [as stop] (C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\shelljs\make.js:36:40)
    at C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\shelljs\make.js:48:27
    at Array.forEach (native)
    at null._onTimeout (C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\shelljs\make.js:46:10)
    at Timer.listOnTimeout (timers.js:92:15)
```

**Comment by jmervine** [comment]
Is that consistent? I've seen that periodically. I kind of just want to get rid of forever all together since we don't use it in production.

**Comment by XhmikosR** [comment]
Yeah, I'm getting that with `node make travis` in the enforce-lf branch.

```
  1946 passing (1m)
  1 failing

  1) functional bootstrap https://maxcdn.bootstrapcdn.com/twitter-bootstrap/2.1.1/js/bootstrap.min.js has integrity:

      AssertionError: 'sha384-v2XCZ4x+SDQreGSr5HtkSOzDeIzE4K5aqWp3xcskLxq7Y5S76C4sqh/MEurBY1fz' == 'sha384-gYjuefvjBnMR9aV2JTKd0GeQnxj+mKMBirUgIH+rBsMaUe1iAC8UtOnqPrZV8s2V'
      + expected - actual

      -sha384-v2XCZ4x+SDQreGSr5HtkSOzDeIzE4K5aqWp3xcskLxq7Y5S76C4sqh/MEurBY1fz
      +sha384-gYjuefvjBnMR9aV2JTKd0GeQnxj+mKMBirUgIH+rBsMaUe1iAC8UtOnqPrZV8s2V

      at C:\Users\xmr\Desktop\bootstrap-cdn\tests\functional_test.js:59:16
      at request (C:\Users\xmr\Desktop\bootstrap-cdn\tests\functional_test.js:44:16)
      at assertSRI (C:\Users\xmr\Desktop\bootstrap-cdn\tests\functional_test.js:55:5)
      at Context.<anonymous> (C:\Users\xmr\Desktop\bootstrap-cdn\tests\functional_test.js:97:21)
      at callFnAsync (C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\mocha\lib\runnable.js:349:8)
      at Test.Runnable.run (C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\mocha\lib\runnable.js:301:7)
      at Runner.runTest (C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\mocha\lib\runner.js:422:10)
      at C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\mocha\lib\runner.js:528:12
      at next (C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\mocha\lib\runner.js:342:14)
      at C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\mocha\lib\runner.js:352:7
      at next (C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\mocha\lib\runner.js:320:5)



C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\shelljs\src\common.js:340
        throw e;
        ^

Error: exec: internal error
    at Object.error (C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\shelljs\src\common.js:67:11)
    at _exec (C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\shelljs\src\exec.js:266:12)
    at C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\shelljs\src\common.js:275:23
    at assertExec (C:\Users\xmr\Desktop\bootstrap-cdn\make.js:22:16)
    at Function.target.suite (C:\Users\xmr\Desktop\bootstrap-cdn\make.js:34:9)
    at Object.global.target.(anonymous function) [as suite] (C:\Users\xmr\Deskto
p\bootstrap-cdn\node_modules\shelljs\make.js:36:40)
    at Function.target.travis (C:\Users\xmr\Desktop\bootstrap-cdn\make.js:71:16)

    at Object.global.target.(anonymous function) [as travis] (C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\shelljs\make.js:36:40)
    at C:\Users\xmr\Desktop\bootstrap-cdn\node_modules\shelljs\make.js:48:27
    at Array.forEach (native)
```

**Comment by XhmikosR** [comment]
Err, scratch that, even on Travis, we get the error thrown.

**Comment by jmervine** [comment]
Right! Because we do use `make start`, we use it for starting a demonized version of the app during functional testing.

**Comment by jmervine** [comment]
I'll need to look at this. I'm thinking that it means the process is crashing before `node make stop` is being run. Which isn't good.

**Comment by XhmikosR** [comment]
Can this be due to a change in recent shelljs version?

**Comment by XhmikosR** [comment]
Confirmed it's definitely an issue with the updated shelljs. Switching back to `^0.5.3` doesn't throw the internal error anymore. There are many issues on their repo  https://github.com/shelljs/shelljs/issues?utf8=%E2%9C%93&q=is%3Aissue%20internal%20error

**Comment by jdorfman** [comment]
@XhmikosR @jmervine this will be looked at next week. Have a good weekend everybody =)

File: make.js [hunk 2]
```diff
@@ -114,7 +124,10 @@ var MOCHA_OPTS = ' --timeout 15000 --slow 500';
                     var res = exec(BOOTLINT + ' -d W013 ' + output);
 
                     echo('+ node make stop');
-                    target.stop();
+                    try {
+                        // it's okay if this fails
+                        target.stop();
+                    } catch (e) { }
```
Comment (jmervine) [inline make.js]: I know this feels hacky, and it kind of is, but honestly it's really doesn't matter if this pass or fails, it has no impact on the bootlint action.

**Comment by XhmikosR** [comment]
@jmervine: this can be simplified to this only.  https://github.com/MaxCDN/bootstrap-cdn/compare/xmr-jmervine-default-production

I can force push your branch if you want.

**Comment by jmervine** [comment]
@XhmikosR I'll add the spacing to app.json, but I want to keep the `try / catch`. There's not reason that failing to stop the app should ever cause `bootlint` to fail.

**Comment by jmervine** [comment]
@jdorfman @XhmikosR looks like the travis integration stopped running here...

**Comment by jmervine** [comment]
Nevermind, there it goes...

File: make.js [hunk 3]
```diff
@@ -72,9 +72,7 @@ var MOCHA_OPTS = ' --timeout 15000 --slow 500';
     };
 
     target.tryStop = function() {
-        try {
-            exec(FOREVER + ' stop app.js');
-        } catch (e) { }
+        exec(FOREVER + ' stop app.js || true');
```
Comment (jmervine) [inline make.js]: @XhmikosR I know this is super annoying, but I'd probably do something similar in bash.

Comment (XhmikosR) [inline make.js]: Yeah, it's an understandable workaround.

**Comment by jmervine** [comment]
@XhmikosR gonna merge unless you have any objections

**Comment by XhmikosR** [comment]
Only thing please fetch and rebase against develop.

I'd like to test it one more time probably tomorrow morning.

**Comment by jmervine** [comment]
fetched, rebased, squashed

File: package.json [hunk 4]
```diff
@@ -33,7 +33,7 @@
     "request": "^2.64.0",
     "serve-favicon": "^2.3.0",
     "serve-static": "^1.10.0",
-    "shelljs": "^0.5.3",
+    "shelljs": "^0.7.2",
```
Comment (XhmikosR) [inline package.json]: This shouldn't be here.

Comment (jmervine) [inline package.json]: I fixed the error with `^0.7.2` on this branch.

**Pull request merged**